### PR TITLE
Add support for Internship classes within a Company class, and json loading support for Internships

### LINKED ISSUE #109: Add in Internship class and support storing Internship data in JSON file
Internships are supposed to be contained within Companies.

A Company can have 0 or more Internships. 

The JSON file must also reflect and support this change.

### PR DESCRIPTION
**Pull request by oeggy03**
Added the new Internship class, along with InternshipDescription, InternshipName and InternshipInterviewDateTime classes to serve as objects representing attributes of the Internship class.

Modified the JsonAdaptedCompany class and added the JsonAdaptedInternship class to support loading of Internship data from addressbook.json.

Edited existing test json data files.

Edited Test classes of Company class to now accomodate internships.

Updated the toString() methods of Company, and Internship and its attributes to make the returned String more readable for debugging.

Added tests for DateTimeParserUtil, JsonAdaptedInternship.

Add tests for AddCompanyCommand and Company classes

Add Internship classes to a package in company and write tests for Internship classes

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/editcommand/EditCompanyCommand.java
Comment (Tim-Siu) [inline src/main/java/seedu/address/logic/commands/editcommand/EditCompanyCommand.java:1]: Great to see you adapted all the current functionalities!

File: src/test/java/seedu/address/model/company/internship/InternshipInterviewDateTimeTest.java
Comment (Tim-Siu) [inline src/test/java/seedu/address/model/company/internship/InternshipInterviewDateTimeTest.java:1]: Testing looks comprehensive to me.

File: src/test/java/seedu/address/testutil/CompanyBuilder.java
Comment (Tim-Siu) [inline src/test/java/seedu/address/testutil/CompanyBuilder.java:1]: Great to see you updated all the utility classes. Amazing!

**Review by Tim-Siu — APPROVED** [review]
Solid contribution and a giant leap forward for the SOCareers! All the existing code and test cases have been updated. The code looks neat and I can hardly find anything for nit-picking. Eager to embrace the new phase of SOCareers.

File: src/main/java/seedu/address/logic/Messages.java [hunk 1]
```diff
@@ -57,16 +58,29 @@ public static String formatPerson(Person person) {
...
         final StringBuilder builder = new StringBuilder();
         builder.append(company.getCompanyName())
-                .append("; Phone: ")
+                .append(";\n Phone: ")
                 .append(company.getCompanyPhone())
-                .append("; Email: ")
+                .append(";\n Email: ")
                 .append(company.getCompanyEmail())
-                .append("; Address: ")
+                .append(";\n Descriptions: ")
```
Comment (lunaroddity) [inline src/main/java/seedu/address/logic/Messages.java:65]: Should this be `Description` instead?

**Pull request merged**